common/src/main/java/io/netty/util/concurrent/DefaultPromise.java

```diff
@@ -666,18 +668,13 @@ private boolean await0(long timeoutNanos, boolean interruptable) throws Interrup
 
         checkDeadLock();
 
-        long startTime = System.nanoTime();
-        long waitTime = timeoutNanos;
-        boolean interrupted = false;
-        try {
-            for (;;) {
-                synchronized (this) {
-                    if (isDone()) {
-                        return true;
-                    }
+        synchronized (this) {
+            boolean interrupted = false;
+            try {
+                while(!isDone() && (timeoutNanos -= System.nanoTime() - startTime) > 0) {
```

Comment:
Fix check-style error.

```suggestion
                while (!isDone() && (timeoutNanos -= System.nanoTime() - startTime) > 0) {
```

Reply:
done it.